aten/src/ATen/native/cpu/UnaryOpsKernel.cpp

```diff
@@ -645,6 +645,21 @@ static void rsqrt_kernel(TensorIterator& iter) {
   }                                                                                              \
   REGISTER_DISPATCH(op##_stub, &op##_kernel)
 
+  #define IMPLEMENT_COMPLEX_STRUCTURED_KERNEL(op)              \
+  static void op##_kernel(TensorIteratorBase& iter) {                         \
+    TORCH_INTERNAL_ASSERT(iter.ntensors() == 2);                              \
+    AT_DISPATCH_FLOATING_AND_COMPLEX_TYPES_AND1(kBFloat16, iter.dtype(), #op "_vml_cpu", [&]() {\
+      cpu_kernel_vec(iter,                                                    \
+        [=](scalar_t a) __ubsan_ignore_undefined__ -> scalar_t {              \
+          return std::op(a);                                                  \
```

Comment:
hmm, just comparing this code to `IMPLEMENT_ITERATOR_LAMBDA` which was there before, it looks like the current version of the unary kernels in this file use `vml` implementations ([here](https://github.com/pytorch/pytorch/blob/59ac0ff03709f6fb690918e0cd5717a0716c2eb6/aten/src/ATen/native/cpu/UnaryOpsKernel.cpp#L619)) of each operator.

Judging from the headers of `Loops.h` ([here](https://github.com/pytorch/pytorch/blob/59ac0ff03709f6fb690918e0cd5717a0716c2eb6/aten/src/ATen/native/cpu/Loops.h#L25)) and `vml.h` ([here](https://github.com/pytorch/pytorch/blob/59ac0ff03709f6fb690918e0cd5717a0716c2eb6/aten/src/ATen/cpu/vml.h#L25)), I think that `cpu_kernel` might only be meant for binary ops, so we should continue using `vml` here.

@ngimel, do you happen to know if `cpu_kernel`/`cpu_kernel_vec` are only meant for binary ops (and not unary ops)? Then we should probably just keep using the existing macro here that calls `vml`, and call `cast_outputs` directly.

Reply:
`cpu_kernel_vec` is used a lot in Activation.cpp for unary ops, e.g. https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/cpu/Activation.cpp#L278-L287, this is clearly a unary op. 
That said, a lot of unary ops go through vml/mkl (mkl is a lot faster for special functions), and we should keep whatever implementation is used now.

Reply:
Got it, thanks. @hameerabbasi sorry for the whiplash- we probably want to keep using the existing macro so we can keep using vml/mkl, and just tack on a call to `cast_outputs()`.

Reply:
Done, tests pass locally, so resolving this.